### PR TITLE
No extra zeros on numericality

### LINKED ISSUE #106: Feature request: Validate leading zero for numericality
# Goal:

* Expect `"010"` to be false in `numericality` validator

# Description:

It will be great if there is an option like `noLeadingZero` in `numericality`.

## Why

* Value from HTML5 number box is string
* The coercion of `+` will force `"010"` to be `10`

Facts above make `010` to be a valid number which it should not be in reality in my opinion.

# JS:

```javascript

var c = {
	"numericality": {
    "greaterThanOrEqualTo":1,
    "lessThanOrEqualTo":100,
    "message":"<id> from 1 to 100",
  },
  "presence": true,
};

// Expect to be false, but get true
console.log(
	'validation result ',
	validate.single('010', c)
);

// No problem
console.log(
	'validation result ',
	validate.single('10', c)
);
```

# Fiddler:

https://jsfiddle.net/8okbtbuz/2/

### PR DESCRIPTION
**Pull request by mondwan**
Fixes #106

### CODE REVIEW

File: specs/validators/numericality-spec.js [hunk 1]
```diff
@@ -293,4 +293,30 @@ describe("validators.numericality", function() {
       , value = "foo";
     expect(numericality(value, options)).toBe(message);
   });
+
+  describe("noExtraZero", function() {
+    it("disallows extra zeros", function () {
+        expect(numericality("1.0000000", {noExtraZero: true})).toBe(
+            "must be no extra zeros"
+        );
+        expect(numericality("1.00", {noExtraZero: true})).toBe(
```
Comment (ansman) [inline specs/validators/numericality-spec.js]: This test and the one above should not fail. Having trailing zeroes should be allowed.

Comment (mondwan) [inline specs/validators/numericality-spec.js]: Refer to the above floating point numbers, I am not sure how should I validate such number?

Should I allow all trailing zeros for floating point numbers?

Comment (ansman) [inline specs/validators/numericality-spec.js]: Yes, but I'll fix that.

**Comment by ansman** [comment]
Merged manually.

**Pull request closed without merging**